### PR TITLE
Add CentOS based nodejs4 stack to default assembly

### PR DESCRIPTION
**Pull request by dharmit**
Signed-off-by: Dharmit Shah <dshah@redhat.com>

### What does this PR do?
Add NodeJS stack based on CentOS to the default assembly

#### Changelog

Add CentOS based nodejs4 stack to default assembly

#### Release Notes

Add CentOS based nodejs4 stack to default assembly

#### Docs PR

cc @l0rd

### CODE REVIEW

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 1]
```diff
@@ -2384,5 +2384,95 @@
...
+              "org.eclipse.che.ssh"
+            ],
+            "servers": {},
+            "attributes": {
+              "memoryLimitBytes": "2147483648"
+            }
+          }
+        },
+        "recipe": {
+          "location": "dharmit/centos-nodejs",
```
Comment (l0rd) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json]: Don't you want to use registry.centos registry here?

**Review by l0rd — CHANGES_REQUESTED** [review]

File: ide/che-core-ide-stacks/src/main/resources/stacks.json (hunk 1 above)
Comment (dharmit) [reply ide/che-core-ide-stacks/src/main/resources/stacks.json]: @l0rd modified.

File: ide/che-core-ide-stacks/src/main/resources/stacks.json [hunk 2]
```diff
@@ -2460,5 +2461,94 @@
       "version": "3.3.9"
     }
   ]
-}
+  },
+  {
+  "id": "CentOS nodejs",
```
Comment (l0rd) [inline ide/che-core-ide-stacks/src/main/resources/stacks.json]: All stacks, except centos nodejs and wildfly swarm have lowercase IDs without spaces. Can you please change the IDs into:
- nodejs4
- wildfly-swarm

**Review by l0rd — CHANGES_REQUESTED** [review]
@dharmit thanks for your update. We have just figured out that there might be an issue with IDs containing spaces. Can you address that too? Thanks for your patience.

**Comment by dharmit** [comment]
@l0rd thanks for pointing that out. I updated the PR. Let me know if it looks good or needs further modification.

**Review by l0rd — APPROVED** [review]

**Comment by l0rd** [comment]
Thanks @dharmit. I'm ok with the current state of this PR. @benoitf do you think we can merge this?

**Review by benoitf — APPROVED** [review]
@l0rd ok for me. Need a PM for release notes (added)

**Review by slemeur — APPROVED** [review]

**Pull request merged**